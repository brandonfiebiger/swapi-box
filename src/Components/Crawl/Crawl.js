import React from 'react';
import PropTypes from 'prop-types';

import './Crawl.css';

const Crawl = (props) => {
  const displayFilms = props.films.map( (film, index) => (
    <div key={film.title + index}>
      <h2>{film.title}</h2>
      <p>{film.crawl}</p>
      <p className='film-end'>{film.date}</p>
    </div>
  ));

  return (
    <section className="crawler-container">
      <div className="crawler">
        {displayFilms}
      </div>
    </section>
  );
};

export default Crawl;

Crawl.propTypes = {
  films: PropTypes.arrayOf(PropTypes.object)
};