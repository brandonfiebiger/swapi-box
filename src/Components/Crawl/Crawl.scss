.crawler-container {
  position: fixed;
  display: flex;
  justify-content: center;
  width: 100%;
  height: 100vh;
  perspective: 300px;
  overflow: hidden;
  background: url(../../assets/background.png) no-repeat;
  z-index: -10;
}

.crawler {
  position: relative;
  font-size: 2rem;
  width: 100%;
  animation: crawl 60s infinite linear;
  color: #F9C835;
  p {
    width: 100%;
  }
  .film-end {
    margin-bottom: 1200px;
  }
}

@keyframes crawl {
  0% {
    top: 100vh;
    transform: rotateX(20deg) translateZ(0);
  }

  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px)

  }
}