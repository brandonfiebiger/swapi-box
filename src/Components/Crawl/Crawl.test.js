import React from 'react';
import { shallow } from 'enzyme';

import Crawl from './Crawl';

describe('Crawl', () => {
  
  it('should match snapshot', () => {
    const films = []
    const wrapper = shallow(<Crawl films={films}/>);
    
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot with one film', () => {
    const films = [{ title: 'test1', crawl: 'text', date: '1'}]
    const wrapper = shallow(<Crawl films={films}/>);
    
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot with multiple films', () => {
    const films = [{ title: 'test1', crawl: 'text', date: '1' }, { title: 'test2', crawl: 'text', date: '2' }]
    const wrapper = shallow(<Crawl films={films}/>);
    
    expect(wrapper.html()).toMatchSnapshot();
  });

});