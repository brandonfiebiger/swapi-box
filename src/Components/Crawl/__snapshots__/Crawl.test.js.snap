// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crawl should match snapshot 1`] = `"<section class=\\"crawler-container\\"><div class=\\"crawler\\"></div></section>"`;

exports[`Crawl should match snapshot with multiple films 1`] = `"<section class=\\"crawler-container\\"><div class=\\"crawler\\"><div><h2>test1</h2><p>text</p><p class=\\"film-end\\">1</p></div><div><h2>test2</h2><p>text</p><p class=\\"film-end\\">2</p></div></div></section>"`;

exports[`Crawl should match snapshot with one film 1`] = `"<section class=\\"crawler-container\\"><div class=\\"crawler\\"><div><h2>test1</h2><p>text</p><p class=\\"film-end\\">1</p></div></div></section>"`;
